anago

```diff
@@ -662,53 +606,27 @@ prepare_tree () {
   # Now set the branch we're on
   branch=$(gitlib::current_branch)
 
-  # If this is a new branch, rev openapi-spec version files
-  # Because this modifies files on master and there's a good chance the
-  # master has moved ahead by now, the tag has to occur before this file
-  # change or the later rebase in gitlib::push_master() will rewrite the
-  # commit associated with the tag and orphan it.
-  if [[ -n "$PARENT_BRANCH" && $label == alpha ]]; then
-    git_tag $label $branch || return 1
-    rev_openapi_versions $label || return 1
-    return 0
-  fi
-
-  # rev openapi-spec version files on branch for beta tags
-  case $label in
-    beta*) rev_openapi_versions $label || return 1 ;;
-  esac
-
-  # COMMENTING OUT FOR NOW AS THE WAY FORWARD IS NOT CLEAR
-  # generate docs on new branches (from master) only
-  # If the entirety of this session is based on a branch from master
-  # (PARENT_BRANCH), and this iteration of prepare_tree() is operating on
-  # the NON-master branch itself, versionize the docs
-  #if [[ "$PARENT_BRANCH" == "master" && "$branch" != "master" ]]; then
-  #  logecho -n "Generating docs for ${RELEASE_VERSION[$label]}: "
-  #  logrun -s $TREE_ROOT/hack/generate-docs.sh || return 1
-  #  logecho -n "Committing: "
-  #  logrun -s git commit -am \
-  #            "Generating docs for ${RELEASE_VERSION[$label]} on $branch."
-  #fi
-
-  # Clear all CHANGELOG-N.NN.md files on the branch.
-  # This replaces the above commented out change to the branch.
-  # Some unique change is required on the branch to allow it to be git
-  # describe'd as something other than the current tag on master
-  # If/when the generate-docs.sh issue is sorted out, this will become
-  # optional, though probably still useful anyway.
+  # Clear all CHANGELOG-N.NN.md files on the branch and insert the
+  # generated one, if # $PARENT_BRANCH==master, which seems to be in the
+  # weeks between branch creation the first official release of the branch.
+  # TODO: can this condition on whether the branch is "beta" or "rc" and be
+  # more simply readable to a maintainer?
   if [[ "$PARENT_BRANCH" == "master" && "$branch" != "master" ]]; then
     logecho -n "Remove any previous CHANGELOG-*.md files: "
     logrun -s git rm -f CHANGELOG-*.md || return 1
     logecho -n "Copy master $CHANGELOG_FILE to $branch branch: "
     logrun -s git checkout master -- $CHANGELOG_FILE || return 1
     logecho -n "Committing deleted CHANGELOG-*.md files: "
-    logrun -s git commit -am \
-              "Delete extraneous CHANGELOG-*.md files on branch." \
-     || return 1
+    logrun -s git commit -am "Delete extraneous CHANGELOG-*.md files on branch."
```

Comment:
did you mean to drop the `|| return 1`? if so why?

Reply:
Yes.  Should've been in the commit comment.

The function as implemented here now unconditionally tags.  Once.  Instead of misc. taggings possible in conditional branches and/or if you fall through the whole function.

Reply:
And it does so _after_ the one piece of code that may modify the repo with desired content for the tag.

Reply:
We could though add a lot of || return 1's all over anago

Reply:
@spiffxp -- Thanks for the catch! It allowed me to find another `logrun` missing a return code.
Fixed both in https://github.com/kubernetes/release/pull/1013/commits/648b267d6085a4c10875224915fe646e86a8c744.

Reply:
>Tagging happens once. 

Which looks fine, it was just that this particular return 1 change seemed unrelated to that.  Thanks for the fixup.

Reply:
Indeed you're right.  I had that conditional block in my head as ending with the return, not being a continuation of the line and conditionally returning only on error.  One of the other blocks was more in the pattern of conditionally do and exit success.